lib/routes/account.js

```diff
@@ -995,6 +1063,11 @@ module.exports = (log, db, mailer, Password, config, customs, checkPassword, pus
             authAt: sessionToken.lastAuthAt()
           }
 
+          // Ensure that the correct verification method is set if one has been specified.
+          if (verificationMethod) {
+            response.verified = false
+            response.verificationMethod = verificationMethod
```

Comment:
For completeness, we should probably also set the `verificationReason` here like we do in the code below.